src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,61 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int BOUND = 5;
+    private static final int MAX_VALUE = 100;
+    private static final String DEFAULT_COLOR = "white";
+    private static final int DEFAULT_RADIUS = 10;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int index = random.nextInt(BOUND);
+        switch (index) {
+            case 0:
+                return getCircle();
+            case 1:
+                return getIsoscelesTrapezoid();
+            case 2:
+                return getSquare();
+            case 3:
+                return getRightTriangle();
+            default:
+                return getRectangle();
+        }
+    }
+
+    private int getRandomInt() {
+        return random.nextInt(MAX_VALUE) + 1;
+    }
+
+    private String getRandomColor() {
+        return colorSupplier.getRandomColor();
+    }
+
+    private Figure getCircle() {
+        return new Circle(getRandomColor(), getRandomInt());
```

Comment:
passing expressions as constructor arguments decreases the code readability

see [here](https://mate-academy.github.io/jv-program-common-mistakes/java-core/abstract-class-interface/oop-advanced#%D1%81reating-a-figure-dont-pass-expressions-in-constructor)

**applicable for all other methods**